modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -2181,22 +2181,24 @@ private static PeersAndLearners configurationFromAssignments(Collection<Assignme
      */
     // TODO: IGNITE-18619 Maybe we should wait here to create indexes, if you add now, then the tests start to hang
     private CompletableFuture<PartitionStorages> getOrCreatePartitionStorages(TableImpl table, int partitionId) {
-        return CompletableFuture
-                .supplyAsync(() -> {
-                    MvPartitionStorage mvPartitionStorage = table.internalTable().storage().getOrCreateMvPartition(partitionId);
-                    TxStateStorage txStateStorage = table.internalTable().txStateStorage().getOrCreateTxStateStorage(partitionId);
+        InternalTable internalTable = table.internalTable();
+
+        MvPartitionStorage mvPartition = internalTable.storage().getMvPartition(partitionId);
+
+        return (mvPartition != null ? completedFuture(mvPartition) : internalTable.storage().createMvPartition(partitionId))
```

Comment:
Can partition be created in a parallel thread at the same time, so that getMvPartition returns null, but at the moment when we call `createMvPartition` we already called it in another thread?

Reply:
Only one thread should create storage, and receive the rest.

On the one hand, we will be able to catch those who are trying to create tables in parallel and reduce the number of code, on the other hand, we can divide it into two methods and use it correctly. Well, or store a map with storage futures and create them lazily.

wdyt?

Reply:
I think we can leave it as it is, but we need a test for this behaviour

Reply:
When I forgot to change the code here, quite a few tests fell on me.